File: backend/src/main/java/com/coing/domain/bookmark/entity/Bookmark.java
package com.coing.domain.bookmark.entity;

import java.time.LocalDateTime;

import com.coing.domain.coin.market.entity.Market;
import com.coing.domain.user.entity.User;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "bookmark", uniqueConstraints = {
	@UniqueConstraint(columnNames = {"user_id", "market_id"})
})
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Builder
public class Bookmark {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "bookmark_id")
	private Long id;

	// 북마크를 등록한 회원
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "user_id", nullable = false)
	private User user;

	// 북마크 대상 마켓 (코인 ID와 동일한 역할을 함; 예: "KRW-BTC")
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "market_id", nullable = false)
	private Market market;

	@Column(nullable = false, name = "create_at")
	private LocalDateTime createAt;

	@Column(name = "update_at")
	private LocalDateTime updateAt;

}

File: backend/src/main/java/com/coing/domain/bookmark/entity/Bookmark.kt
package com.coing.domain.bookmark.entity

import com.coing.domain.coin.market.entity.Market
import com.coing.domain.user.entity.User
import com.coing.util.BaseEntity
import jakarta.persistence.*
import org.hibernate.annotations.OnDelete
import org.hibernate.annotations.OnDeleteAction

@Entity
@Table(
    name = "bookmark",
    uniqueConstraints = [UniqueConstraint(columnNames = ["user_id", "market_id"])]
)
class Bookmark(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "bookmark_id")
    val id: Long? = null,

    // 북마크를 등록한 회원
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    val user: User,

    // 북마크 대상 마켓
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "market_id", nullable = false)
    val market: Market

) : BaseEntity()
